packages/components/src/tools-panel/stories/index.tsx

```diff
@@ -162,93 +189,143 @@ export const WithNonToolsPanelItems = () => {
-						<UnitControl
+							onDeselect={ () => setMinWidth( undefined ) }
+							isShownByDefault={
+								isFirstToolsPanelItemShownByDefault
+							}
+						>
+							<UnitControl
+								label="Minimum width"
+								value={ minWidth }
+								onChange={ ( next ) => setMinWidth( next ) }
+							/>
+						</SingleColumnItem>
+						<SingleColumnItem
+							hasValue={ () => !! width }
 							label="Width"
-							value={ width }
-							onChange={ ( next ) => setWidth( next ) }
-						/>
-					</SingleColumnItem>
-					<SingleColumnItem
-						hasValue={ () => !! height }
-						label="Height"
-						onDeselect={ () => setHeight( undefined ) }
-						isShownByDefault={ false }
-					>
-						<UnitControl
+							onDeselect={ () => setWidth( undefined ) }
+							isShownByDefault={ false }
+						>
+							<UnitControl
+								label="Width"
+								value={ width }
+								onChange={ ( next ) => setWidth( next ) }
+							/>
+						</SingleColumnItem>
+						<SingleColumnItem
+							hasValue={ () => !! height }
 							label="Height"
-							value={ height }
-							onChange={ ( next ) => setHeight( next ) }
-						/>
-					</SingleColumnItem>
-				</ToolsPanel>
-			</Panel>
-		</PanelWrapperView>
+							onDeselect={ () => setHeight( undefined ) }
+							isShownByDefault={ false }
+						>
+							<UnitControl
+								label="Height"
+								value={ height }
+								onChange={ ( next ) => setHeight( next ) }
+							/>
+						</SingleColumnItem>
+					</ToolsPanel>
+				</Panel>
+			</PanelWrapperView>
+
+			<button
+				onClick={ () =>
+					setIsFirstToolsPanelItemShownByDefault(
+						! isFirstToolsPanelItemShownByDefault
+					)
+				}
+				aria-pressed={
+					isFirstToolsPanelItemShownByDefault ? 'true' : 'false'
+				}
+				style={ {
+					marginTop: '2rem',
+				} }
+			>
+				{ isFirstToolsPanelItemShownByDefault
+					? 'Make first PanelItem hidden by default'
+					: 'Make first PanelItem shown by default' }
+			</button>
+		</>
 	);
 };
 
 WithOptionalItemsPlusIcon.args = {
-	isShownByDefault: false,
+	...Default.args,
+	label: 'Tools Panel (optional items only)',
 };
 
 const { Fill: ToolsPanelItems, Slot } = createSlotFill( 'ToolsPanelSlot' );
-const panelId = 'unique-tools-panel-id';
 
-export const WithSlotFillItems = () => {
-	const [ attributes, setAttributes ] = useState( {} );
+export const WithSlotFillItems: ComponentStory< typeof ToolsPanel > = ( {
+	resetAll: resetAllProp,
+	panelId,
+	...props
+} ) => {
+	const [ attributes, setAttributes ] = useState< {
+		width?: string;
+		height?: string;
+	} >( {} );
 	const { width, height } = attributes;
 
-	const resetAll = ( resetFilters = [] ) => {
+	const resetAll: typeof resetAllProp = ( resetFilters = [] ) => {
 		let newAttributes = {};
 
 		resetFilters.forEach( ( resetFilter ) => {
 			newAttributes = {
 				...newAttributes,
+				// TODO: the `ResetFilter` type doesn't specify any attributes
+				// and doesn't return any objects
+				// @ts-ignore
```

Comment:
This is something that I wasn't able to fix on my own. According to the type defs, `ResetAllFilter` is of type `() => void;`, and therefore is not supposed to accept arguments.

We have two alternatives here:
- either change the `ResetAllFilter` type to accept arguments
- or change the next line to `...resetFilter()`

This same issue happens a few times across the file

Reply:
Thanks for catching this one 👍 

In terms of which option, I believe the type is incorrect as prior uses of the `resetAllFilter` are to pass through the current data so each filter in turn was aware of the latest changes and wouldn't override updates from previous filters.

If I recall correctly, the color supports panel uses a different batch approach that might allow the second option you presented. 

Given there might already be external use of the ToolsPanel, I think I'd still lean towards fixing the type to match existing uses. Also, the ToolsPanel wasn't typed initially and was converted to TypeScript after the fact, which might further point to the types being incorrect.

Reply:
> I think I'd still lean towards fixing the type to match existing uses

Ok! I tried to find a good way to describe the `ResetAllFIlter` type , but the fact that it could literally receive any data as an argument A(or undefined), and return any data (or `void`) made me lean towards using `( attributes?: any ) => any` (https://github.com/WordPress/gutenberg/pull/47944/commits/364fb8727711c9bded7678d9dfc833520dfd0859)

(@mirka or @tyxla , do you have any suggestions on how to type this callback more strictly ?)

I've also updated the README in both `ToolsPanel` and `ToolsPanelItem` to reflect those changes (https://github.com/WordPress/gutenberg/pull/47944/commits/08e55bc9cc712d747e3bd1ea6957a890b29c8cf5)

@aaronrobertshaw , what do you think? If this looks good, I'll let you decide what pattern we should advocate for in example snippets (and potentially change usages in the repo to follow that style too).

Reply:
> what do you think?

I think this is still an improvement over the incorrect type not matching the callback's intended use. 

Unfortunately, I'm short on ideas for typing it more strictly. I'd suggest the general expectation by consumers would be that the same type would be maintained as the data gets passed through each `resetAllFilter`. I don't know how we'd achieve that, though. 

> I'll let you decide what pattern we should advocate for in example snippets (and potentially change usages in the repo to follow that style too).

We have two use cases around resetting values, and only one needs the reset all filters.
1. When the ToolsPanel is created, it has a fixed set of child `ToolsPanelItems`, which aren't rendered via slot fills into the panel. This means the `ToolsPanel` can know what needs to be reset or executed for the `resetAll` function and can avoid individual items needing to define `resetAllFilter`. The best examples here are the Global Styles panels in the site editor e.g. [Dimensions Panel](https://github.com/WordPress/gutenberg/blob/ba1a2ba66f208915edf59f78abd05c4cfc1f60b3/packages/edit-site/src/components/global-styles/dimensions-panel.js#L407-L420).
2. The `ToolsPanel` renders a slot that may contain an unknown number of child `ToolsPanelItems`. In this case, the consumer of the `ToolsPanel` needs to rely on each `ToolsPanelItem` specifying how to effect its reset via the `resetAllFilter` callback. The primary examples of this are the [block support panels](https://github.com/WordPress/gutenberg/blob/8675f86f885ca389f3bca39971b22cb897bab61b/packages/block-editor/src/components/inspector-controls/block-support-tools-panel.js#L24) in the block editor.

Outside of our editors and their block support panels, I believe the first use case above would be the most common. We'd probably be best served to lean towards demonstrating it in our JSDoc example snippets. The Storybook is there to demonstrate the more edge case use of `SlotFills` with the `ToolsPanel`.

I'll push a commit shortly that might form the basis for our example snippet.

Reply:
It might also be worth noting that the passing of the `attributes` arg to the `resetAllFilter` only came about due to needing to reset only specific properties within nested objects, e.g. only radius, and not color, width, and style, within a `style.border` attribute. 

Keeping any example use of the `ToolsPanel` simple sounds like a win. If a consumer's use case is more complicated the extra Storybook examples provide direction.

Reply:
Agreed, let's showcase the simple usage (ie. without `resetAllFilter`) in our example snippets